tools/odbc/parameter_wrapper.cpp

```diff
@@ -0,0 +1,151 @@
+#include "parameter_wrapper.hpp"
+#include "duckdb_odbc.hpp"
+#include "duckdb/common/types/decimal.hpp"
+
+using duckdb::Decimal;
+using duckdb::idx_t;
+using duckdb::ParameterDescriptor;
+using duckdb::ParameterWrapper;
+
+ParameterWrapper::~ParameterWrapper() {
+}
+
+void ParameterWrapper::Clear() {
+	param_descriptors.clear();
+	cur_paramset_idx = 0;
+}
+
+void ParameterWrapper::ResetParamSetIndex() {
+	cur_paramset_idx = 0;
+}
+
+SQLRETURN ParameterWrapper::GetValues(std::vector<Value> &values) {
+	values.clear();
+	if (param_descriptors.empty()) {
+		return SQL_NO_DATA;
+	}
+	D_ASSERT(cur_paramset_idx < paramset_size);
+	// Fill values
+	for (idx_t desc_idx = 0; desc_idx < param_descriptors.size(); ++desc_idx) {
+		// set a proper parameter value
+		auto ret = param_descriptors[desc_idx].SetValue(cur_paramset_idx);
+		if (param_status_ptr) {
+			param_status_ptr[cur_paramset_idx] = ret;
+		}
+		if (ret != SQL_PARAM_SUCCESS) {
+			std::string error_msg;
+			error_msg.reserve(100);
+			auto len = snprintf((char *)error_msg.data(), 100,
```

Comment:
Perhaps use `StringUtil::Format` here? We try to avoid C style functions where possible.